### ISSUE TITLE
Garbage in overscan area not displayed

### ISSUE THREAD

**Issue by retrorepair** [ekeeke/Genesis-Plus-GX#220]
On a real Geneis/Megadrive, the VDP will spit out garbage (sprite data?) in the overscan area, usually the bottom right.

This is evedent in lots of games (all of them?) and if you load a game in the Exodus emulator you can see it there, as well as on real hardware but although Genesis Plus GX does draw the overscan area (borders) it doesn't show this garbage.

More a nit picky thing really but I feel it adds to the overall feeling of playing on a real console as I always remember this from my childhood. In the name of accuracy.....

**Comment by ekeeke** [comment]
Actually, this happens whenever the game tries to write the color palette (CRAM), this is not garbage sprite data but the color value that is being written. It happens because of bus access conflict with the current pixel readout. And it does not happen only in border area but can also occurs in active screen area if the game tries to update CRAM mid-screen, it's just more common (and more visible) in border area.

It was initially not emulated because:
- it increases emulation processing requirements and requires cycle-accurate CRAM writes
- it's more a limitation of the original hardware than really a feature
- it's the kind of stuff most people never paid attention to when they were playing with their original console back in time (I certainly didn't and I had a PAL console were the borders were quite visible compared to NTSC ones where they barely were) and which suddenly become 'important' for em  ulators to 'feel right'.

That's said, there are some recent homebrew demos that kinda use this 'feature' to display pictures in 'direct-color' mode (by disabling the display then doing precisely timed and continuous DMA to CRAM) so emulating this could actually be interesting. That's why I will leave this request open as it might eventually be added sometime in the future (by me or someone else)

**Comment by NTxC** [comment]
Please consider working on this feature. This is the only feature that's missing from Genesis Plus GX that is implemented in Blastem. If you implement this my life will be complete.

**Comment by SN902** [comment]
Yes, it's like a vinyl without any minor crackles, if you make dots artefacts available as custom option in retroarch for android and windows, my life will be complete.

**Comment by getCursorsExe** [comment]
If it increases emulation processing requriements, then how about CRAM Dots as an option?

**Comment by getCursorsExe** [comment]
This explains why the 512 COLORS TITANIC RECORD in TiTAN Overdrive Beta is displayed as horizontal lines. This is probably also because of the emulator unknowingly displaying only 16 colors in one scanline.

**Comment by nad53** [comment]
I can totally relate to this subject. If there's something I have seen that is missing on Genesis plus GX it's the garbage pixels as the author just mentioned. It may not bring up useful image information for the game but it's an actual nostalgic feature that helps even beyond for perfect, if not the almost-perfect original experience emulation.

This thread is 3 years old but I hope at some point it can be released this feature for GPGX as well. It's really hard for me to notice any difference with Blastem, but this particular feature for CRAM pixels actually brings up another accuracy boost for original looking experience.
Outstanding work in your core to be honest, really appreciate what you have done to this point with this core.

**Comment by retrorepair** [comment]
Is this still a possibility?

**Comment by levonnn** [comment]
Still hoping for this one.
